xrpl4j-client/src/main/java/org/xrpl/xrpl4j/client/XrplClient.java

```diff
@@ -252,6 +259,140 @@ public FeeResult fee() throws JsonRpcClientErrorException {
+   */
+  protected Optional<? extends TransactionResult<? extends Transaction>> getValidatedTransaction(
+    final Hash256 transactionHash
+  ) {
+    Objects.requireNonNull(transactionHash);
+
+    try {
+      final TransactionResult<? extends Transaction> transactionResult = this.transaction(
+        TransactionRequestParams.of(transactionHash), Transaction.class
+      );
+      return Optional.ofNullable(transactionResult).filter(TransactionResult::validated);
+    } catch (JsonRpcClientErrorException e) {
+      // The transaction was not found on ledger, warn on this, but otherwise return.
+      LOGGER.warn(e.getMessage(), e);
+      return Optional.empty();
+    }
+  }
+
+  /**
+   * Enumeration of the types of responses from isFinal function.
+   */
+  public enum FinalStatus {
+    /**
+     * Transaction has not been validated until this moment, which means it could be validated in the
+     * following closed ledgers, or, the XRPL peer you are connected to could be missing the ledgers
+     * it would have been validated on.
+     */
+    NOT_FINAL,
+    /**
+     * Transaction is final/validated on the ledger successfully.
+     */
+    VALIDATED_SUCCESS,
+    /**
+     * Transaction is final/validated on the ledger with some failure.
+     */
+    VALIDATED_FAILURE,
+    /**
+     * The lastLedgerSequence of the tx has passed, i.e., transaction was not included in any closed
+     * ledgers ranging from the one it was submitted on to the lastLedgerSequence.
+     */
+    EXPIRED
+  }
+
+  /**
+   * Check if the transaction is final on the ledger or not.
+   *
+   * @param transactionHash {@link Hash256} of the submitted transaction to check the status for.
+   * @param submittedOnLedgerIndex {@link LedgerIndex} on which the transaction with hash transactionHash
+   *        was submitted. This can be obtained from submit() response of the tx as validatedLedgerIndex.
+   * @param lastLedgerSequence The ledger index/sequence of type {@link UnsignedInteger} after which the
+   *        transaction will expire and won't be applied to the ledger.
+   * @param sequence The sequence number of the account submitting the {@link Transaction}. A {@link Transaction}
+   *        is only valid if the Sequence number is exactly 1 greater than the previous transaction
+   *        from the same account.
+   * @param account The unique {@link Address} of the account that initiated this transaction.
+   * @return {@code true} if the {@link Transaction} is final/validated else {@code false}.
+   */
+  public FinalStatus isFinal(
+    Hash256 transactionHash,
+    LedgerIndex submittedOnLedgerIndex,
+    UnsignedInteger lastLedgerSequence,
+    UnsignedInteger sequence,
+    Address account
+  ) {
+
+    return getValidatedTransaction(transactionHash)
+      .map(transactionResult -> {
+        //Do some stuff with transactionResult and return a FinalityStatus
+        String resultCode = transactionResult.metadata()
```

Comment:
Let's introduce a new `Finality` object so that we can capture the resultCode when making this call:

```

  public interface Finality {
    FinalityStatus finalityStatus();
    
    Optiona<String> resultCode();

    @Default
    default String resultCodeSafe(){
      return resultCode().orElseThrow(...);
    }
  }
```